src/transformers/trainer.py

```diff
@@ -1092,13 +1118,24 @@ def _save_tpu(self, output_dir: Optional[str] = None):
         if xm.is_master_ordinal():
             os.makedirs(output_dir, exist_ok=True)
             torch.save(self.args, os.path.join(output_dir, "training_args.bin"))
+            json.dump(
+                self.log_history, open(os.path.join(output_dir, "log_history.json"), "w"), indent=2, ensure_ascii=False
+            )
 
         # Save a trained model and configuration using `save_pretrained()`.
         # They can then be reloaded using `from_pretrained()`
         if not isinstance(self.model, PreTrainedModel):
             raise ValueError("Trainer.model appears to not be a PreTrainedModel")
 
         xm.rendezvous("saving_checkpoint")
+        # Storing the number of floating-point operations that went into the model
```

Comment:
Those 7 lines are duplicated, maybe put them in a private method to refactor a bit?

Reply:
agreed, done